pkg/auth/handler/webapp/authflowv2/login.go

```diff
@@ -89,6 +100,20 @@ func (h *AuthflowV2LoginHandler) ServeHTTP(w http.ResponseWriter, r *http.Reques
 		return
 	}
 
+	if r.URL.Query().Get(webapp.PreviewQueryKey) == webapp.PreviewModeInline {
+		var previewHandler handlerwebapp.PreviewHandler
```

Comment:
PreviewHandler is redundant. Or it provides no useful abstraction. Please remove it.

I suggest we add a method .Preview to AuthflowController. AuthflowController provides an abstraction of error handling.

```go
if IsPreViewModeInline(r) {
  // .Preview will take care of the returned error, and render the error page.
  h.Controller.Preview(func() error {
     // ...
  })
}
```

Reply:
What about
```
// HandlePreview will check the query, and return without doing anything if the query parameter is not "inline"
h.Controller.HandleInlinePreview(func() error { ... })

h.Controller.HandleStep(...)
```

Reply:
something like?

```go
func (h *Handler) handler(r, w) {
  h.Controller.InlinePreview(func() error {
     ...
  })
  h.Controller.Get(func() error {
    ...
  })
  ...
  if isPreviewModeInline(r) {
    h.Controller.HandlePreview()
    return;
  }
  h.Controller.StartAuthFlow()
}
```

Reply:
@tung2744 @louischan-oursky See if this implementation looks good to you two

https://github.com/authgear/authgear-server/pull/4359/commits/e03e1c4bbe735ff12ea699f6ed4df15f8513233a

Reply:
Looks good to me.